code/game/objects/structures/ladders.dm

```diff
@@ -24,6 +29,9 @@
 		update_icon()
 
 /obj/structure/ladder/update_icon()
+	if(istype(src, /obj/structure/ladder/dive_point))

```

Comment:
They have pre-defined sprites that are static. This was to prevent them from turning back into ladders.